trail/upload/serializers.py

```diff
@@ -0,0 +1,34 @@
+from .models import Metadata
+from rest_framework import serializers
+
+
+class DynamicMetadataSerializer(serializers.ModelSerializer):
+    """
+    A ModelSerializer that takes an additional `fields` argument that
+    controls which fields should be displayed.
+    """
+
+    def __init__(self, *args, **kawrgs):
+        fields = kawrgs.pop("fields", None)
+
+        super().__init__(*args, **kawrgs)
+
+        if fields is not None:
+            allowed = set(fields)
+            existing = set(self.fields)
+            for field_name in existing - allowed:
+                self.fields.pop(field_name)
+
+
+class MetadataSerializer(DynamicMetadataSerializer):
+    """
+    A MetadataSerializer that takes an additional `fields` argument that
+    controls which fields should be displayed.
+    """
```

Comment:
Same as above. Remember, we inherit the init, except in this case we can also direct users to which key names are allowed as well.

```
Arguments
---------------
fields :  `list[str]`
    List of `Metadata` fields that will be serialized in the response.
```